pilot/pkg/xds/ads.go

```diff
@@ -589,9 +618,30 @@ func (s *DiscoveryServer) pushConnection(con *Connection, pushEv *Event) error {
 	// Send pushes to all generators
 	// Each Generator is responsible for determining if the push event requires a push
 	for _, w := range getPushResources(con.proxy.WatchedResources) {
-		err := s.pushXds(con, pushRequest.Push, currentVersion, w, pushRequest)
```

Comment:
consider

```go
if !features.EnableFlowControl {
   err := s.pushXds(con, pushRequest.Push, currentVersion, w, pushRequest)
   if err != nil {
      return err
   }
  continue
}
```
followed by the new logic... assuming flow control is enabled.

Reply:
make sense

Reply:
:+1: